src/KafkaW/MetaDataQuery.h

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include <chrono>
+#include <librdkafka/rdkafkacpp.h>
+#include <set>
+#include <string>
+
+namespace KafkaW {
+
+using time_point = std::chrono::system_clock::time_point;
+using duration = std::chrono::system_clock::duration;
+
+// \brief Query the broker for topic + partition offset of a given timestamp.
+//
+// \param[in] Broker    Address of one or several brokers.
+// \param[in] Topic     Name of topic.
+// \param[in] Partition Partition id.
+// \param[in] Time      Timestamp for which the offset should be found.
+// \param[in] TimeOut   The amount of time to wait for the meta-data call to
+// complete.
+// \return              The closest offset with a timestamp before Time (if
+// available).
+// \throw MetadataException if a failure occurs.
+std::vector<std::pair<int, int64_t>>
+getOffsetForTime(std::string const &Broker, std::string const &Topic,
+                 std::vector<int> const &Partitions, time_point Time,
+                 duration TimeOut);
+
+// \brief Query the broker for partitions available on a specified topic.
+//
+// \param[in] Broker    Address of one or several brokers.
+// \param[in] Topic     Name of topic.
+// \param[in] TimeOut   The amount of time to wait for the meta-data call to
+// complete.
+// \return              A set of partition id:s.
+// \throw               MetadataException if a failure occurs.
```

Comment:
Don't add throws, the chances of this comment being updated if the code changes is almost 0.

Reply:
I do not really follow your line of reasoning here. Can you provide a reference or example where this is a problem?

Reply:
Go through the code-base and see how many doxy comments are wrong or out-dated, if it is zero then ignore me.
But it is in our code standards not to use throw: https://confluence.esss.lu.se/pages/viewpage.action?pageId=265299412 ;)